dockerfiles/base/scripts/base/library.sh

```diff
@@ -627,3 +337,130 @@ find_and_print_ports_as_notok() {
   error "Ports required to run $CHE_MINI_PRODUCT_NAME are used by another program."
   return 2;
 }
+
+container_exist_by_name(){
+  docker inspect ${1} > /dev/null 2>&1
+  if [ "$?" == "0" ]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+get_server_container_id() {
+  log "docker inspect -f '{{.Id}}' ${1}"
+  docker inspect -f '{{.Id}}' ${1} 2>&1 || false
+}
+
+container_is_running() {
+  if [ "$(docker ps -qa -f "status=running" -f "id=${1}" | wc -l)" -eq 0 ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+wait_until_container_is_running() {
+  CONTAINER_START_TIMEOUT=${1}
+
+  ELAPSED=0
+  until container_is_running ${2} || [ ${ELAPSED} -eq "${CONTAINER_START_TIMEOUT}" ]; do
+    log "sleep 1"
+    sleep 1
+    ELAPSED=$((ELAPSED+1))
+  done
+}
+
+has_compose() {
+  hash docker-compose 2>/dev/null && return 0 || return 1
+}
+
+docker_compose() {
+#  debug $FUNCNAME
+
+  if has_compose; then
+    docker-compose "$@"
+  else
+    docker_run -v "${CHE_HOST_INSTANCE}":"${CHE_CONTAINER_INSTANCE}" \
+                  $IMAGE_COMPOSE "$@"
+  fi
+}
+
+start_test_server() {
+  export AGENT_INTERNAL_PORT=80
+  export AGENT_EXTERNAL_PORT=32768
+
+  # Start mini httpd server to run simulated tests
+  docker run -d -p $AGENT_EXTERNAL_PORT:$AGENT_INTERNAL_PORT --name fakeagent \
+             ${BOOTSTRAP_IMAGE_ALPINE} httpd -f -p $AGENT_INTERNAL_PORT -h /etc/ >> "${LOGS}"
+
+  export AGENT_INTERNAL_IP=$(docker inspect --format='{{.NetworkSettings.IPAddress}}' fakeagent)
+  export AGENT_EXTERNAL_IP=$CHE_HOST
+}
+
+stop_test_server() {
+  # Remove httpd server
+  docker rm -f fakeagent >> "${LOGS}"  
+}
+
+test1() {
```

Comment:
should we drop some of code of the network info command as it seems we have a lot of duplicates
also the function name test1() is hard to know what is really done there and for which part this test is.